### PR TITLE
TCOMP-2119 : Avro record to contains arrays with null

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-2119
Allow array to have null elements.
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

**Review by undx — APPROVED** [review]
LGTM w/ a comment about unfinished test.

File: component-runtime-beam/src/test/java/org/talend/sdk/component/runtime/beam/spi/record/AvroRecordBuilderTest.java [hunk 1]
```diff
@@ -258,6 +274,55 @@ void recordWithNewSchema() {
...
+        JsonEncoder encoder = EncoderFactory.get().jsonEncoder(avroRecord.getSchema(), outputArray, true);
+        writer.write(avroRecord, encoder);
+        encoder.flush();
+        String chain = new String(outputArray.toByteArray(), StandardCharsets.UTF_8);
+        /*
+         * JsonObject jsonObject = Json.createReader(new StringReader(chain)).readObject();
+         * JsonArray jsonArray = jsonObject.getJsonObject("farray").getJsonArray("array");
+         * Assertions.assertEquals(3, jsonArray.size());
+         * Assertions.assertEquals(2, jsonArray.stream().filter(JsonObject.class::isInstance).count());
+         */
```
[the remark this answers is not in the document]
Comment (clesaec) [reply component-runtime-beam/src/test/java/org/talend/sdk/component/runtime/beam/spi/record/AvroRecordBuilderTest.java]: updated

File: component-runtime-beam/src/test/java/org/talend/sdk/component/runtime/beam/coder/registry/SchemaRegistryCoderTest.java [hunk 2]
```diff
@@ -87,13 +87,14 @@ void avoidNPE() {
                 .newEntryBuilder()
                 .withName("createdBy")
                 .withType(Schema.Type.RECORD)
-                .withElementSchema(factory.newSchemaBuilder(Schema.Type.STRING).build())
+                .withElementSchema(factory.newSchemaBuilder(Schema.Type.STRING).build()) // element schema for record
+                                                                                         // means nothing ?
```
[the remark this answers is not in the document]
Comment (clesaec) [reply component-runtime-beam/src/test/java/org/talend/sdk/component/runtime/beam/coder/registry/SchemaRegistryCoderTest.java:91]: ElementSchema is useful only for Array Type, here, we define a Record with an Element Schema ... It means nothing.

**Review by undx — APPROVED** [review]
LGTM

**Review by fdesiles — APPROVED** [review]
Approving comment update.

**Pull request merged**